### ISSUE TITLE
Easy way to join a bunch of subspaces and rooms

### ISSUE THREAD

**Issue by HarHarLinks** [vector-im/element-web#17388]
#### Is your suggestion related to a problem? Please describe.
When joining a space, it is rather tedious having to join a room one by one, each join taking a minute or two.

#### Describe the solution you'd like.
![image](https://user-images.githubusercontent.com/2803622/118826254-9d286f00-b8bb-11eb-92cf-b00bcf4013cf.png)

obviously you have the 3 existing buttons only as space admin, so most often there's only the join button.

**Comment by t3chguy** [comment]
The issue here, given each join can take such a long time due to server performance issues, this would take a very long time and be very jarring as rooms slowly pop into your room list

**Comment by HarHarLinks** [comment]
This same effect is already happening if you click join, and then click away from the loading screen. The fact that I have to take the time and repeat the step dozens of times when I join a medium to large space like matrix community only makes the issue worse.

I faintly remember something about immediate feedback about this or a similar case in the issues before.
If possible, I suggest immediately adding the room(s) to the list upon join, with some kind of "this room is still loading" indication, and removing it again (to historical?) if joining failed.

**Comment by jaiwanth-v** [comment]
Related #17352

**Comment by HarHarLinks** [comment]
I must have missed #17352 when opening this issue, however it makes some good points I want to mention
- have a un/tick all selection box to facilitate joining all
- if joining >1 room this way, you stay on the space overview page

further regarding performance issues:
- when selecting "many"(?) to join here, show a popup "warning this may take some time, continue?" "abort" "continue" "continue and don't warn again"

**Comment by mattdm** [comment]
I would like this too. I would specifically like each space to have a list of subspaces and rooms that are auto-joined when the space is joined -- just like currently there's a list of suggestions. (Keep that too, just add an additional level.)

**Comment by dkasak** [comment]
> The issue here, given each join can take such a long time due to server performance issues, this would take a very long time and be very jarring as rooms slowly pop into your room list

I don't think this is that jarring, as long as it doesn't mess with controls and focus. If the room list is updating in the background, I personally wouldn't care much about that. Though the ideal solution would probably be to add a placeholder for the room immediately like HarHarLinks suggested.

**Comment by dkasak** [comment]
My proposed solution (from a closed duplicate issue), in which I suggested modifying the "Explore rooms" screen of a space:

> I propose that the "Join" button for subspaces in a space shouldn't immediately switch to the joined subspace. It should instead replace the "Join" button with a spinner while it's performing the action and then with the "View" button (or other completion signal) when it's done. The subspace would then appear in the space side bar and the user would switch to it manually if they want to explore it further.
